cloud/services/container/nodepools/reconcile.go

```diff
@@ -44,20 +45,32 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+// setReadyStatusFromConditions updates the GCPManagedMachinePool's ready status based on its conditions.
+func (s *Service) setReadyStatusFromConditions() {
+	machinePool := s.scope.GCPManagedMachinePool
+	if conditions.IsTrue(machinePool, clusterv1.ReadyCondition) || conditions.IsTrue(machinePool, infrav1exp.GKEMachinePoolUpdatingCondition) {
```

Comment:
Where we set the ready condition to true we can also set the `Status.Ready`.

Reply:
this is what this helper function does, if the ready condition is true then we set the ready status to true

Reply:
🤣 I understand what the helper function is doing, my comment was more around not using this function and instead setting status ready at the same time you set tge condition to true.

This function is defereed and also i'm not a fan on logic gitting off of conditions.

Reply:
This is more a personal/stylistic preference so lets not block on this.

Reply:
I see. The reason why I refactored it this way is because there were some places where we forgot to set the ready status when setting the ready condition. I thought this would make it more reliable so future contributors don't have to worry about setting both at the same time.